mcp/sensors/star_camera_transmit.c

```diff
@@ -0,0 +1,396 @@
+}
+
+// To be called after we pack the packet so we make sure that
+//  no rogue parameters are hanging around to muck it up
+static void clear_command_data_sc2() {
+    memset(&CommandData.sc2_commands, 0, sizeof(CommandData.sc2_commands));
+}
+
+// To be called after transmitting the packet to make sure that
+// we don't have old packet data sticking around (in charge??)
+static void clear_packet_data_sc1() {
+    memset(&sc1_command_packet, 0, sizeof(sc1_command_packet));
+}
+
+// To be called after transmitting the packet to make sure that
+// we don't have old packet data sticking around (in charge??)
+static void clear_packet_data_sc2() {
+    memset(&sc2_command_packet, 0, sizeof(sc2_command_packet));
+}
+
+// points the talker thread at the right status boolean to watch
+static int check_sc_thread_bool(int which) {
+    if (which == 1) {
+        return CommandData.sc_bools.sc1_command_bool;
+    } else if (which == 2) {
+        return CommandData.sc_bools.sc2_command_bool;
+    } else {
+        blast_err("Invalid star camera to check thread bool %d", which);
+        return -1;
+    }
+}
+
+// points the talker thread at the right commands to watch
+static int check_sc_send_commands(int which) {
+    if (which == 1) {
+        return CommandData.sc1_commands.send_commands;
+    } else if (which == 2) {
+        return CommandData.sc2_commands.send_commands;
+    } else {
+        blast_err("Invalid star camera to check commands %d", which);
+        return -1;
+    }
+}
+
+static int prepare_packet(int which) {
+    if (which == 1) {
+        return prepare_command_packet_sc1();
+    } else if (which == 2) {
+        return prepare_command_packet_sc2();
+    } else {
+        blast_err("Invalid star camera to pack packets %d", which);
+        return -1;
+    }
+}
+
+static void clear_structs(int which) {
+    if (which == 1) {
+        clear_command_data_sc1();
+        clear_packet_data_sc1();
+        return;
+    } else if (which == 2) {
+        clear_command_data_sc2();
+        clear_packet_data_sc2();
+        return;
+    } else {
+        blast_err("Invalid star camera to clear structs %d", which);
+        return;
+    }
+}
+
+static int check_for_reset(int which) {
+    if (which == 1 && CommandData.sc_resets.reset_sc1_comm) {
+        // reset the flag
+        CommandData.sc_resets.reset_sc1_comm = 0;
+        return 1;
+    } else if (which == 2 && CommandData.sc_resets.reset_sc2_comm) {
+        CommandData.sc_resets.reset_sc2_comm = 0;
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+static void reset_command_bools(void) {
+    CommandData.sc_bools.sc1_command_bool = 1;
+    CommandData.sc_bools.sc2_command_bool = 1;
+}
+
+// our omnibus function for vomiting the packets to MCP
+void *star_camera_command_thread(void *args) {
+    struct socket_data * socket_target = args;
+    static int first_time = 1;
+    int sockfd;
+    struct addrinfo hints, *servinfo, *servinfoCheck;
```

Comment:
Put variable declarations, especially if they're of different type, on separate lines.